config/config_test.go

```diff
@@ -14,6 +14,105 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestExternalCacheURLValidate(t *testing.T) {
+	type testOut struct {
+		expectError  bool
+		expectedHost string
+		expectedPath string
+	}
+
+	testCases := []struct {
+		desc string
+		data ExternalCache
+		out  testOut
+	}{
+		{
+			"With http://",
+			ExternalCache{Host: "http://www.google.com", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "/path/v1"},
+		},
+		{
+			"Without http://",
+			ExternalCache{Host: "www.google.com", Path: "/path/v1"},
+			testOut{expectError: false, expectedHost: "www.google.com", expectedPath: "/path/v1"},
+		},
+		{
+			"No scheme but '//' prefix",
+			ExternalCache{Host: "//www.google.com", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "/path/v1"},
+		},
+		{
+			"// appears twice",
+			ExternalCache{Host: "//www.google.com//", Path: "path/v1"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "//path/v1"},
+		},
+		{
+			"Host has an only // value",
+			ExternalCache{Host: "//", Path: "path/v1"},
+			testOut{expectError: true, expectedHost: "path", expectedPath: "/v1"},
+		},
+		{
+			"only scheme host, valid path",
+			ExternalCache{Host: "http://", Path: "/path/v1"},
+			testOut{expectError: true, expectedHost: "", expectedPath: "/path/v1"},
+		},
+		{
+			"No host, path only",
+			ExternalCache{Host: "", Path: "path/v1"},
+			testOut{expectError: true, expectedHost: "", expectedPath: "path/v1"},
+		},
+		{
+			"No host, nor path",
+			ExternalCache{Host: "", Path: ""},
+			testOut{expectError: false, expectedHost: "", expectedPath: ""},
+		},
+		{
+			"Invalid http at the end",
+			ExternalCache{Host: "www.google.com", Path: "http://"},
+			testOut{expectError: true, expectedHost: "www.google.com", expectedPath: "/http://"},
+		},
+		{
+			"Scheme longer than host",
```

Comment:
Not sure this is really testing the case of "Scheme longer than host", but rather "Unknown scheme included"

Reply:
Please change this description to "Host has an unknown scheme"